Assets/MixedRealityToolkit/InputModule/Scripts/Utilities/Interactions/TwoHandManipulatable.cs

```diff
@@ -0,0 +1,631 @@
+            {
+                targetScale = m_scaleLogic.Update(m_handsPressedLocationsMap);
+            }
+
+            HostTransform.position = targetPosition;
+            HostTransform.rotation = targetRotation;
+            HostTransform.localScale = targetScale;
+        }
+
+        private void OnOneHandMoveUpdated()
+        {
+            var targetPosition = m_moveLogic.Update(m_handsPressedLocationsMap.Values.First(), HostTransform.position);
+
+            HostTransform.position = targetPosition;
+
+#if !TODO_ROTATE_FACE_USER
+            // TODO: Rotate object to face user as needed.
+            TryRotateObjectToFaceUser();
+#endif
+
+        }
+
+        private void OnTwoHandManipulationEnded()
+        {
+        }
+
+        private Vector3 GetHandsCentroid()
+        {
+            Vector3 result = m_handsPressedLocationsMap.Values.Aggregate(Vector3.zero, (current, state) => current + state);
+            return result / m_handsPressedLocationsMap.Count;
+        }
+
+        private void OnTwoHandManipulationStarted(State newState)
+        {
+            if ((newState & State.Rotating) > 0)
+            {
+                m_rotateLogic.Setup(m_handsPressedLocationsMap, HostTransform);
+            }
+            if ((newState & State.Moving) > 0)
+            {
+                m_moveLogic.Setup(GetHandsCentroid(), HostTransform);
+            }
+            if ((newState & State.Scaling) > 0)
+            {
+                m_scaleLogic.Setup(m_handsPressedLocationsMap, HostTransform);
+            }
+        }
+
+        private void OnOneHandMoveStarted()
+        {
+            Assert.IsTrue(m_handsPressedLocationsMap.Count == 1);
+
+            m_moveLogic.Setup(m_handsPressedLocationsMap.Values.First(), HostTransform);
+        }
+        private void OnManipulationStarted()
+        {
+            if (StartedManipulating != null)
+            {
+                StartedManipulating();
+            }
+            InputManager.Instance.PushModalInputHandler(gameObject);
+
+            // Show Bounding Box visual on manipulation interaction
+            if(GlobalBoundingBox != null)
+            {
+                HasBoundingBox = true;
+            }
+        }
+        private void OnManipulationEnded()
+        {
+            if (StoppedManipulating != null)
+            {
+                StoppedManipulating();
+            }
+            InputManager.Instance.PopModalInputHandler();
+
+            // Hide Bounding Box visual on release
+            if (GlobalBoundingBox != null)
+            {
+                HasBoundingBox = false;
+            }
+        }
+    }
+
+    /// <summary>
+    /// RotateLogic implements common logic for rotating holograms using a handlebar metaphor. 
+    /// When a manipulation starts, call Setup.
+    /// Call Update any time to update the move logic and get a new rotation for the object.
+    /// </summary>
+    public class HandlebarRotateLogic
```

Comment:
public classes should be in their own files.

Reply:
Fixed.